### DISCUSSION TITLE
VS Code break the file

### DISCUSSION THREAD

**Discussion by fahdshaykh** [microsoft/vscode-discussions#788] [Extension Development QnA] [answered]
Hi team, today i am working on local machine on windows, i was working on a project in Laravel that is php file and then suddenly show file corrupt and showing that "the file is not displayed in the text editor because it is eith binary or uses an unsupported text encoding"
now i am confused what that happened it was due to vs code or something else ?
![Screenshot (9)](https://github.com/microsoft/vscode-discussions/assets/86095746/8557ba38-8934-48a3-95cf-03e17f7fb383)
![Screenshot (10)](https://github.com/microsoft/vscode-discussions/assets/86095746/965f26db-224a-4870-a363-922a0d004ef5)
![Screenshot (11)](https://github.com/microsoft/vscode-discussions/assets/86095746/d278f2e9-4733-417d-9a91-6bf9bd8b1fa3)

**Comment by gjsjohnmurray** [comment]
It is very unlikely that what you are seeing is being caused by VS Code. I suggest you try opening your PHP files in another editor such as Notepad so you can check whether they have been corrupted.

In any case, this forum is intended for developers who are creating extensions for VS Code.

**Reply by fahdshaykh** [reply]
@gjsjohnmurray i tried to opened in notepad++ it show something unreadable. I am still wondering it may be git issue or vs code ?
![broken cart file](https://github.com/microsoft/vscode-discussions/assets/86095746/cd37d322-96cf-4e3e-ac81-001db492059c)

**Reply by gjsjohnmurray** [reply, marked as answer]
Hardware problem? Virus? If you come up with a set of steps to reproduce VS Code causing this data loss in your files please report it at https://github.com/microsoft/vscode/issues

**Reply by fahdshaykh** [reply]
@gjsjohnmurray
thanks it was due to virus attack.
